components/ws-manager-mk2/controllers/create.go

```diff
@@ -615,6 +615,9 @@ func createWorkspaceEnvironment(sctx *startWorkspaceContext) ([]corev1.EnvVar, e
 	memoryInMegabyte := res.Memory().Value() / (1000 * 1000)
 	result = append(result, corev1.EnvVar{Name: "GITPOD_MEMORY", Value: strconv.FormatInt(memoryInMegabyte, 10)})
 
+	cpuCount := res.Cpu().MilliValue() / 1000
+	result = append(result, corev1.EnvVar{Name: "GITPOD_CPU_COUNT", Value: strconv.FormatInt(int64(cpuCount), 10)})
```

Comment:
This'll work well, I think. Its the requests, which'll work for Enterprise and PAYG (with PAYG there is burst to go higher than requests), but we're not talking about limits here.

Reply:
I assume you're referring to `g1-standard`, good catch.

Perhaps it would make sense to also retrieve the workspace class limits, and use the value of cpu.min.

Reply:
I have tested with different values (worked) and added a feature flag, so we can slowly roll out.